src/ejabberd_local.erl

```diff
@@ -411,57 +388,26 @@ directed_to(<<>>, _) ->
 directed_to(_, _) ->
     user.
 
--spec update_table() -> ok | {atomic|aborted, _}.
-update_table() ->
-    case catch mnesia:table_info(iq_response, attributes) of
-        [id, module, function] ->
-            mnesia:delete_table(iq_response);
-        [id, module, function, timer] ->
-            ok;
-        {'EXIT', _} ->
-            ok
-    end.
-
--spec get_iq_callback(ID :: id()) -> 'error' | {'ok', Mod :: atom(), fun() | atom()}.
+-spec get_iq_callback(ID :: id()) -> error | {ok, fun()}.
 get_iq_callback(ID) ->
-    case mnesia:dirty_read(iq_response, ID) of
-        [#iq_response{module = Module, timer = TRef,
-                      function = Function}] ->
-            cancel_timer(TRef),
-            mnesia:dirty_delete(iq_response, ID),
-            {ok, Module, Function};
+    case ets:lookup(?IQRESPONSE, ID) of
+        [{ID, Callback, TRef}] ->
+            erlang:cancel_timer(TRef),
```

Comment:
This makes me think, the ping IQs timeouts are handled twice here... c2s also registers its own. Well This PR solves _not using mnesia_, which is a huge improvement, and everything else actually doesn't change, so it seems fine. But hell, there's an important tech debt waiting here. Let's keep the PRs small and work on that on the next one.